### PR TITLE
Telegraf 1.21

### PR DESCRIPTION
**Pull request by noramullen1**

### CODE REVIEW

File: data/telegraf_plugins.yml [hunk 1]
```diff
@@ -968,6 +975,13 @@ input:
     introduced: 1.3.0
     tags: [linux, macos, windows, systems]
 
+  - name: Logical Volume Manager
+    id: lvm
+    description: |
+      The Logical Volume Manager collects information about physical volumes, volume groups, and logical volumes in Linux.
+    introduced: 1.21.0
+    tags: [linux]
```
Comment (sanderson) [inline data/telegraf_plugins.yml:983]:
```suggestion
    tags: [linux, systems]
```

File: data/telegraf_plugins.yml [hunk 2]
```diff
@@ -1328,6 +1342,13 @@ input:
     introduced: 1.5.0
     tags: [linux, macos, windows, applications]
 
+  - name: OpenStack
+    id: openstack
+    description: |
+      The OpenStack input plugin This plugin collects metrics on [OpenStack services](https://www.openstack.org/).
```
Comment (sanderson) [inline data/telegraf_plugins.yml:1348]:
```suggestion
      The OpenStack input plugin collects metrics on [OpenStack services](https://www.openstack.org/).
```

File: data/telegraf_plugins.yml [hunk 3]
```diff
@@ -2149,6 +2177,13 @@ output:
     introduced: 1.0.0
     tags: [linux, macos, windows, logging]
 
+  - name: GroundWork
+    id: groundwork
+    description: |
+      The  Graylog output plugin writes to a [GroundWork Monitor](https://www.gwos.com/product/groundwork-monitor/) instance.
```
Comment (sanderson) [inline data/telegraf_plugins.yml:2183]:
```suggestion
      The Groundwork output plugin writes to a [GroundWork Monitor](https://www.gwos.com/product/groundwork-monitor/) instance.
```

File: data/telegraf_plugins.yml [hunk 4]
```diff
@@ -2051,6 +2072,13 @@ output:
     introduced: 1.20.0
     tags: [linux, macos, windows, cloud, datastores]
 
+  - name: Azure Event Hubs
+    id: event_hubs
+    description: |
+      The Azure Data Explorer output plugin writes metrics to a single Azure Event Hub within an [Event Hubs](https://azure.microsoft.com/en-gb/services/event-hubs/) namespace.
+    introduced: 1.21.0
+    tags: [linux, macos, windows, cloud, datastores]
```
Comment (sanderson) [inline data/telegraf_plugins.yml:2080]:
```suggestion
    tags: [linux, macos, windows, cloud, data-stores]
```

File: content/telegraf/v1.21/about_the_project/release-notes-changelog.md [hunk 5]
```diff
@@ -0,0 +1,3369 @@
...
+- Update `eclipse/paho.mqtt.golang` module from 1.3.0 to 1.3.5.
+- Shutdown Telegraf gracefully on Windows Service.
+- Skip `knxlistener` when writing the sample configuration file.
+- Update `nats-sever` to support `openbsd`.
+- Revert unintented corruption of the Makefile.
+- Filter client certificates by DNS names.
+- Update `etc/telegraf.conf` and `etc/telegraf_windows.conf`.
+- Add full metadata to configuration for `common.kafka`.
+- Update `google.golang.org/grpc` module from 1.39.1 to 1.40.0.
+
```
Comment (sanderson) [inline content/telegraf/v1.21/about_the_project/release-notes-changelog.md]:
```suggestion
- Print loaded plugins and deprecations for `--once` and `--test` flags.
- Filter client certificates by DNS names.
- Add full metadata to configuration for `common.kafka`.
- Update `eclipse/paho.mqtt.golang` module from 1.3.0 to 1.3.5.
- Update `nats-sever` to support `openbsd`.
- Update `etc/telegraf.conf` and `etc/telegraf_windows.conf`.
- Update `google.golang.org/grpc` module from 1.39.1 to 1.40.0.

### Bug fixes
- Restart Telegraf service if it's already running and upgraded via RPM.
- Revert unintended corruption of the Makefile.
- Shutdown Telegraf gracefully when running as a Windows service.
- Skip `knxlistener` when writing the sample configuration file.

```

File: content/telegraf/v1.21/about_the_project/release-notes-changelog.md [hunk 6]
```diff
@@ -0,0 +1,3369 @@
...
+## Input plugin updates
```
Comment (sanderson) [inline content/telegraf/v1.21/about_the_project/release-notes-changelog.md]:
```suggestion
### Plugin updates

#### Input plugin updates
```

File: content/telegraf/v1.21/about_the_project/release-notes-changelog.md [hunk 7]
```diff
@@ -0,0 +1,3369 @@
...
+- Directory Monitor (`directory_monitor`):
+  - Fix to when when data format is CSV and `csv_skip_rows>0` and `csv_header_row_count>=1`.
+  - Adds the ability to create and name a tag containing the filename.
+- ElasticSearch (`elasticsearch_query`): Add debug query output.
+- HTTP Listener v2: (`http_listener_v2`): Fix panic on close to check that Telegraf is closing.
+- Kubernetes Inventory (`kube_inventory`): Set TLS server name configuration properly.
+- Modbus (`modbus`): Update connection settings (serial).
+- MQTT Consumer (`mqtt_consumer`):
+  - Extracting no longer requires all three fields
+  - Enable extracting tag values from MQTT topics
```
Comment (sanderson) [inline content/telegraf/v1.21/about_the_project/release-notes-changelog.md]:
```suggestion
  - Extracting no longer requires all three fields.
  - Enable extracting tag values from MQTT topics.
```

File: content/telegraf/v1.21/about_the_project/release-notes-changelog.md [hunk 8]
```diff
@@ -0,0 +1,3369 @@
...
+  - Update to use gosmi.
+  - Remove `snmptranslate` from READme and fix default path.
+  - Merge tables with different indexes.
+- StatsD (`statsd`): Fix parse error.
+- Sysstat (`sysstat`): Use unique temporary file.
+- Windows Performance Counters (`win_perf_counters`): Add setting to ignore localization.
+- Windows Services (`win_services`): Add exclude filter.
+- ZFS (`zfs`): Pool detection and metrics gathering for ZFS >= 2.1.x
+
+## Output plugin updates
```
Comment (sanderson) [inline content/telegraf/v1.21/about_the_project/release-notes-changelog.md]:
```suggestion
#### Output plugin updates
```

File: content/telegraf/v1.21/about_the_project/release-notes-changelog.md [hunk 9]
```diff
@@ -0,0 +1,3369 @@
...
+  - Failing test due to port already in use.
+  - Mute UDP/TCP tests by marking them as integration.
+  - TLS support and message format.
+  - Add TCP support.
+- HTTP (`http`): Add `use_batch_format`.
+- InfluxDB V2 (`influxdb_v2`): Add retry to 413 errors with InfluxDB output.
+- Wavefront (`wavefront`): Flush sender on error to clean up broken connections.
+
+
+## Parser plugin updates
```
Comment (sanderson) [inline content/telegraf/v1.21/about_the_project/release-notes-changelog.md]:
```suggestion
#### Parser plugin updates
```

File: content/telegraf/v1.21/about_the_project/release-notes-changelog.md [hunk 10]
```diff
@@ -0,0 +1,3369 @@
...
+- XPath (`xpath`): Handle duplicate registration of protocol-buffer files gracefully
+- JSON v2 (`json_v2`):
+  - Parser timestamp setting order.
+  - Remove dead code.
+  - Support defining field/tag tables within an object table.
+
+## Processor plugin updates
```
Comment (sanderson) [inline content/telegraf/v1.21/about_the_project/release-notes-changelog.md]:
```suggestion
#### Processor plugin updates
```

File: content/telegraf/v1.21/about_the_project/release-notes-changelog.md [hunk 11]
```diff
@@ -0,0 +1,3369 @@
...
+- IfName (`ifname`):
+  - Eliminate MIB dependency.
+  - Parallelism fix.
+  - Add more details to log messages.
+- Starlark (`starlark`): Example for processing `sparkplug_b` messages.
+- RegEx (`regex`): Extend to allow renaming of measurements, tags, and fields.
+
+## Aggregator plugin updates
```
Comment (sanderson) [inline content/telegraf/v1.21/about_the_project/release-notes-changelog.md]:
```suggestion
#### Aggregator plugin updates
```

File: content/telegraf/v1.21/about_the_project/release-notes-changelog.md [hunk 12]
```diff
@@ -0,0 +1,3369 @@
...
+- Implement deprecation infrastructure
+- Add support of aggregator as Starlark script
+
+## New plugins
```
Comment (sanderson) [inline content/telegraf/v1.21/about_the_project/release-notes-changelog.md]:
```suggestion
### New plugins
```

I'd move the "New plugins" section up above "Plugin updates".

File: content/telegraf/v1.21/about_the_project/release-notes-changelog.md [hunk 13]
```diff
@@ -0,0 +1,3369 @@
...
+
+### Inputs
```
Comment (sanderson) [inline content/telegraf/v1.21/about_the_project/release-notes-changelog.md]:
```suggestion
#### Inputs
```

File: content/telegraf/v1.21/about_the_project/release-notes-changelog.md [hunk 14]
```diff
@@ -0,0 +1,3369 @@
...
+
+- [Intel PMU Input Plugin](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/intel_pmu/README.md)(`intel_pmu`) - Contributed by [@bkoltowski](https://github.com/bkotlowski).
+- [Logical Volume Manager Input Plugin](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/lvm/README.md)(`lvm`) - Contributed by @InfluxData.
+- [OpenStack Input Plugin](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/openstack)(`openstack`) - Contributed by [@singamSrikar].(https://github.com/singamSrikar).
+
+### Outputs
```
Comment (sanderson) [inline content/telegraf/v1.21/about_the_project/release-notes-changelog.md]:
```suggestion
#### Outputs
```

File: content/telegraf/v1.21/about_the_project/release-notes-changelog.md [hunk 15]
```diff
@@ -0,0 +1,3369 @@
...
+- [Azure Event Hubs Output Plugin](https://github.com/influxdata/telegraf/blob/master/plugins/outputs/event_hubs/README.md)(`event_hubs`) - Contributed by [@tomconte](https://github.com/tomconte).
+- [GroundWork Output Plugin](https://github.com/influxdata/telegraf/blob/master/plugins/outputs/groundwork/README.md)(`groundwork`) - Contributed by [@VladislavSenkevich)(https://github.com/VladislavSenkevich).
+- [MongoDB Output Plugin](https://github.com/influxdata/telegraf/blob/master/plugins/outputs/mongodb/README.md)(`mongodb`) - Contributed by [@bustedware](https://github.com/bustedware).
+
+### Aggregator
```
Comment (sanderson) [inline content/telegraf/v1.21/about_the_project/release-notes-changelog.md]:
```suggestion
#### Aggregator
```

File: content/telegraf/v1.21/administration/commands.md [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+## Commands
+
+
+
+| Command   | Description                                    |
+| :-------- | :--------------------------------------------- |
+| `config`  | Print out full sample configuration to stdout. |
+| `version` | Print version to stdout.                       |
+
+## Flags
```
Comment (sanderson) [inline content/telegraf/v1.21/administration/commands.md:30]:
```suggestion
## Flags {id="telegraf-command-flags"}
```

This change will fix #3476.

File: content/telegraf/v1.21/administration/enterprise-plugins.md [hunk 17]
```diff
@@ -0,0 +1,19 @@
+---
```
Comment (sanderson) [inline content/telegraf/v1.21/administration/enterprise-plugins.md:1]: I'm guessing we can probably just remove this doc. We can either do it in this PR or with Telegraf 1.22.0

**Review by sanderson — CHANGES_REQUESTED** [review]
Looks good. Just added some suggestions.

File: content/telegraf/v1.21/about_the_project/release-notes-changelog.md [hunk 18]
```diff
@@ -0,0 +1,3369 @@
...
+  - /telegraf/v1.20/reference/release-notes/influxdb/
+menu:
+  telegraf_1_21:
+
+    name: Release notes
+    weight: 10
+    parent: About the project
+---
+## v1.21 [2021-12-15]
+
```
Comment (sjwang90) [inline content/telegraf/v1.21/about_the_project/release-notes-changelog.md]: can you make sure you add the note about RPM at the top? Maybe just

`The signing for RPM digest has changed to use `sha256` to improve security. Due to this change, RPM builds may not be compatible with RHEL6 and older releases (Telegraf only supports releases in RHEL production)`

File: data/telegraf_plugins.yml (hunk 3 above)
Comment (lwandzura) [reply data/telegraf_plugins.yml:2183]: see Scott's suggestion

**Comment by lwandzura** [comment]
@noramullen1  looks good, Scott made some good suggestions, I added a comment

File: data/telegraf_plugins.yml [hunk 19]
```diff
@@ -758,6 +758,13 @@ input:
     introduced: 1.16.0
     tags: [linux, macos, windows, data-stores]
 
+  - name: Intel PMU
+    id: intel_pmu
+    description: |
+      The Intel PMU input plugin exposes Intel PMU (Performance Monitoring Unit) metrics available through Linux Perf subsystem.
+    introduced: 1.21.0
+    tags: [linux]
```
Comment (sjwang90) [inline data/telegraf_plugins.yml:766]: add `systems`

File: data/telegraf_plugins.yml (hunk 1 above)
Comment (sjwang90) [inline data/telegraf_plugins.yml:983]: add `systems`

File: data/telegraf_plugins.yml (hunk 3 above)
Comment (Hipska) [reply data/telegraf_plugins.yml:2183]: It's still not fixed though..

Comment (sjwang90) [reply data/telegraf_plugins.yml:2183]: @Hipska Thanks for checking - opened a PR here https://github.com/influxdata/docs-v2/pull/3803

**Pull request merged**